src/GoodsList.tsx

```diff
@@ -8,7 +8,11 @@ type Props = {
 export const GoodsList: React.FC<Props> = ({ goods }) => (
```

Comment:
It's possible to avoid unnecessary rerender here with the help of React.memo and arePropsEqual function (check whether your prevGoods and nextGoods have the same length and every id from prevGoods corresponds to the id in nextGoods)